docs/how-to-guides/running-feast-in-production.md

```diff
@@ -141,6 +202,145 @@ prediction = model.predict(feature_vector)
 ```
 
 {% hint style="success" %}
-It is important to note that both the training pipeline and model serving service only needs read access to the feature registry and associated infrastructure. This prevents clients from accidentally making changes to the feature store.
+It is important to note that both the training pipeline and model serving service need only read access to the feature registry and associated infrastructure. This prevents clients from accidentally making changes to the feature store.
 {% endhint %}
 
+## 4. Retrieving online features for prediction
+
+Once you have successfully loaded (or in Feast terminology materialized) your data from batch sources into the online store, you can start consuming features for model inference.
+There are three approaches for that purpose sorted from the most simple one (in an operational sense) to the most performant (benchmarks to be published soon):
+
+### 4.1. Use the Python SDK within an existing Python service
+
+This approach is the most convenient to keep your infrastructure as minimalistic as possible and avoid deploying extra services.
+The Feast Python SDK will connect directly to the online store (Redis, Datastore, etc), pull the feature data, and run transformations locally (if required).
+The obvious drawback is that your service must be written in Python to use the Feast Python SDK. 
+A benefit of using a Python stack is that you can enjoy production-grade services with integrations with many existing data science tools.
+
+To integrate online retrieval into your service use the following code:
+```python
+from feast import FeatureStore
+
+with open('feature_refs.json', 'r') as f:
+    feature_refs = json.loads(f)
+
+fs = FeatureStore(repo_path="production/")
+
+# Read online features
+feature_vector = fs.get_online_features(
+    feature_refs=feature_refs,
+    entity_rows=[{"driver_id": 1001}]
+).to_dict()
+```
+
+### 4.2. Consume features via HTTP API from Serverless Feature Server
+
+If you don't want to add the Feast Python SDK as a dependency, or your feature retrieval service is written in a non-Python language,
+Feast can deploy a simple feature server
+on serverless infrastructure (eg, AWS Lambda, Google Cloud Function) for you.
```

Comment:
btw nit: the plan is to use google cloud run since that allows you to run docker images